messaging/src/main/java/nl/nn/adapterframework/extensions/kafka/KafkaListener.java

```diff
@@ -0,0 +1,197 @@
+/*
+   Copyright 2023 WeAreFrank!
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+package nl.nn.adapterframework.extensions.kafka;
+
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+
+import javax.annotation.Nonnull;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.clients.consumer.OffsetAndMetadata;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.serialization.ByteArrayDeserializer;
+import org.apache.kafka.common.serialization.StringDeserializer;
+
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.Setter;
+import nl.nn.adapterframework.configuration.ConfigurationException;
+import nl.nn.adapterframework.core.IListener;
+import nl.nn.adapterframework.core.IPullingListener;
+import nl.nn.adapterframework.core.ListenerException;
+import nl.nn.adapterframework.core.PipeLineResult;
+import nl.nn.adapterframework.core.PipeLineSession;
+import nl.nn.adapterframework.receivers.RawMessageWrapper;
+import nl.nn.adapterframework.stream.Message;
+import nl.nn.adapterframework.stream.MessageContext;
+import nl.nn.adapterframework.util.StringUtil;
+
+/**
+ * Experimental {@link IListener} for listening to a topic in
+ * a Kafka instance.
+ */
+public class KafkaListener extends KafkaFacade implements IPullingListener<ConsumerRecord<String, byte[]>> {
+
+	private static final Predicate<String> TOPIC_NAME_PATTERN = Pattern.compile("^[a-zA-Z0-9._\\-*]*$").asPredicate();
+
+	/** The group id of the consumer */
+	private @Setter String groupId;
+
+	/** Whether to start reading from the beginning of the topic. */
+	private @Setter boolean fromBeginning = false;
+
+	/** How often to check for new topics when using Patterns. (in MS) */
+	private @Setter int patternRecheckInterval = 5000;
```

Comment:
I've learned lately that it is a good idea to put the unit in the name. In every location where you read the code, it is clear without thinking what it means. So, in this case `patternRecheckIntervalMS` or a variation of that is a good idea.

Reply:
Dat wordt wel ook overgenomen als attribuut, wil je dat dan ook? Als variable naam snap ik het nog wel...

Reply:
Ik weet het niet, daarom heb ik dit (nog) niet gedaan. We zijn redelijk inconsistent met wat voor units dit soort time-delay velden hebben.
Een "MS" in de naam maakt het misschien ook voor een attribuut wel duidelijker dat het "MilliSeconds" zijn. Maar een "S" voor "Seconds" is misschien niet erg duidelijk?

Reply:
Vaak gebruikt men `Millis` en 'Secs` of `Seconds` dacht ik. Mij betreft hoeven we het niet groots te veranderen. Nu kwam ik het gewoon tegen. Denk dat het een goede gewoonte is.